ZenPacks/zenoss/ZenPackLib/lib/base/ComponentBase.py

```diff
@@ -231,6 +231,13 @@ def get_containing_relname(self):
         """Return name of containing relationship."""
         for relname, relschema in self._relations:
             if issubclass(relschema.remoteType, ToManyCont):
+                # skip if it's not the real containing rel
+                # for instance an inherited os.filesystems for sub-classed FileSystem
+                rel = getattr(self, relname, None)
```

Comment:
I think the only way to work around the issue and have a valid schema is to purge the other containing relationship from _relations when a new one is added by ModelTypeFactory.  At least then there is a schema that makes sense, and shouldn't break anything. But conceptually it's still weird- i don't know why we would have the ZPLified objects live in a non-default location like that.   Unless we can justify that, we really should be thinking about how to fix this so things live in a normal location.  Not an easy thing to fix now, but it will only get harder the longer we leave it like that.  It's possible that nobody but doug is using these proxies, and in that case we may still be able to put the genie back in the bottle..

I would say that ModelTypeFactory (and possibly ClassSpec as well) need to enforce a few things about relationships:
   - warning or error when changing the schema of an inherited relationship name - this is technically allowable, but almost certainly a bad idea.
   - error when creating a doubly-contained class - this is probably not technically allowed, and definitely a bad idea.